tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/scheduler/OCLNVIDIAGPUScheduler.java

```diff
@@ -61,13 +63,22 @@ public void calculateLocalWork(final TaskMetaData meta) {
                 localWork[2] = calculateGroupSize(calculateEffectiveMaxWorkItemSizes(meta)[2], meta.getGlobalWork()[2], 3);
                 localWork[1] = calculateGroupSize(calculateEffectiveMaxWorkItemSizes(meta)[1], meta.getGlobalWork()[1], 3);
                 localWork[0] = calculateGroupSize(calculateEffectiveMaxWorkItemSizes(meta)[0], meta.getGlobalWork()[0], 3);
+
```

Comment:
I prefer this to be in a different method. One method called `calculateLocalWork` will only calculate the thread block sizes. 

Then, another method will check validity and adapt the local dimensions.